app/scripts/lib/transaction/hooks/delegation-7702-publish.ts

```diff
@@ -165,97 +186,89 @@ export class Delegation7702PublishHook {
-
-    return [userExecution, transferExecution];
+  ): Promise<boolean> {
+    try {
+      const state = (await this.#messenger.call(
+        'BridgeStatusController:getState',
+      ));
+      const gasIncluded7702 =
+        state?.submissionRequests?.[transactionMeta.chainId]?.[transactionMeta.id]
+          ?.quoteResponse?.quote?.gasIncluded7702;
+      return Boolean(gasIncluded7702);
+    } catch (error) {
+      log(
+        'Failed to determine whether the quoted transaction is intended to be gasless through EIP-7702',
+        error,
+      );
+      return false;
+    }
   }
 
-  async #buildDelegation(
+  #buildUnsignedDelegation(
+    environment: DeleGatorEnvironment,
     transactionMeta: TransactionMeta,
-    gasFeeToken: GasFeeToken,
-  ): Promise<Delegation> {
-    const { chainId, txParams } = transactionMeta;
-    const { from } = txParams;
-
-    const caveats = this.#buildCaveats(txParams, gasFeeToken);
+    gasFeeToken: GasFeeToken | undefined,
+    includeTransfer: boolean,
+  ): UnsignedDelegation {
+    const caveats = this.#createCaveats(
+      environment,
+      transactionMeta,
+      gasFeeToken,
+      includeTransfer,
+    );
 
     log('Caveats', caveats);
 
-    const delegation: UnsignedDelegation = {
-      authority: ROOT_AUTHORITY,
+    const delegation = createDelegation({
+      from: transactionMeta.txParams.from as Hex,
+      to: transactionMeta.txParams.from as Hex,
       caveats,
-      delegate: ANY_BENEFICIARY,
-      delegator: from as Hex,
-      salt: new Date().getTime(),
-    };
-
-    log('Unsigned delegation', delegation);
-
-    const signature = await signDelegation({
-      chainId,
-      delegation,
-      from: from as Hex,
-      messenger: this.#messenger,
     });
 
-    log('Delegation signature', signature);
+    log('Delegation', delegation);
 
-    return {
-      ...delegation,
-      signature,
-    };
+    return delegation;
   }
 
-  #buildCaveats(
-    txParams: TransactionParams,
-    gasFeeToken: GasFeeToken,
+  #createCaveats(
+    environment: DeleGatorEnvironment,
+    transactionMeta: TransactionMeta,
+    gasFeeToken: GasFeeToken | undefined,
+    includeTransfer: boolean,
   ): Caveat[] {
-    const { amount, recipient, tokenAddress } = gasFeeToken;
+    const caveatBuilder = createCaveatBuilder(environment);
+
+    const { txParams } = transactionMeta;
     const { data, to } = txParams;
-    const tokenAmountPadded = add0x(remove0x(amount).padStart(64, '0'));
-    const enforcer = process.env.GASLESS_7702_ENFORCER_ADDRESS as Hex;
-
-    const enforcerTerms = add0x(
-      (
-        [
-          tokenAddress,
-          recipient,
-          tokenAmountPadded,
-          to,
-          data ?? EMPTY_HEX,
-        ] as Hex[]
-      )
-        .map(remove0x)
-        .join(''),
-    );
 
-    return [
-      {
-        enforcer,
-        terms: enforcerTerms,
-        args: EMPTY_HEX,
-      },
-    ];
+    if (includeTransfer && gasFeeToken !== undefined) {
```

Comment:
We should have thrown before this point if there was no `gasFeeToken` but a transfer was needed, can we validate that above and remove the `undefined` type here?